Update for downlink sequence counter rollover

itziardelatorre opened the pull request:
### Description

<!--
    Required
    Add here detailed changes summary, testing results, dependencies
    Good example: https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html (Pull request template)
-->
Fix issue related to downlink sequence counter rollover in LoRaWAN.
For the test, the downlink sequence counter of the downlink messages has been incremented by 1600 (< MAX_FCNT_GAP) several times. For example, until 64005 and following situation was shown:
1. Downlink sequence counter increased to 64005 (by 1600 steps)
The radio module accepts this, as expected.
2. After that, downlink sequence counter increased to 65536 (rollover for 32 bits sequence counter)
The radio module rejects this frame

With the change everything works fine.


### Pull request type

<!--
    Required
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front
    of them would change the meaning incorrectly. The only changes to be made are to add a description text under the
    description heading and to add a 'x' to the correct box.
-->
    [ x ] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [ ] Test update
    [ ] Breaking change

### Reviewers

<!--
    Optional
    Request additional reviewers with @username
-->

### Release Notes

<!--
    Optional
    In case of breaking changes, functionality changes or refactors, please add release notes here. 
    For more information, please see [the contributing guidelines](https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html#pull-request-types).
-->

Commit:
Cellular: call AT+CGEREP after sim is ready

Current implementation did call AT+CGEREP before sim was ready
and it was failing in most modems.

itziardelatorre committed:
Bug related to downlink sequence counter rollover

itziardelatorre committed:
Fix issue related to downlink sequence counter rollover in LoRaWAN.
For the test, the downlink sequence counter of the downlink messages has been incremented by 1600 (< MAX_FCNT_GAP) several times. For example, until 64005 and following situation was shown:

Downlink sequence counter increased to 64005 (by 1600 steps)
The radio module accepts this, as expected.
After that, downlink sequence counter increased to 65536 (rollover for 32 bits sequence counter)
The radio module rejects this frame
With the change everything works fine.

0xc0170 committed:
Merge pull request #10081 from jarvte/move_call_CGEREP

Cellular: call AT+CGEREP after sim is ready

0xc0170 reviewed (changes requested):
Expand the commit msg (you provided details in the comment) - about the fix. It's bugfix rather "Bug"

itziardelatorre reviewed (commented):
Fix issue related to downlink sequence counter rollover in LoRaWAN.
For the test, the downlink sequence counter of the downlink messages has been incremented by 1600 (< MAX_FCNT_GAP) several times. For example, until 64005 and following situation was shown:

Downlink sequence counter increased to 64005 (by 1600 steps)
The radio module accepts this, as expected.
After that, downlink sequence counter increased to 65536 (rollover for 32 bits sequence counter)
The radio module rejects this frame
With the change everything works fine.

itziardelatorre committed:
Merge branch 'master' of https://github.com/itziardelatorre/mbed-os

itziardelatorre committed:
Fix issue related to downlink sequence counter rollover in LoRaWAN.

For the test, the downlink sequence counter of the downlink messages has been incremented by 1600 (< MAX_FCNT_GAP) several times. For example, until 64005 and following situation was shown:

- Downlink sequence counter increased to 64005 (by 1600 steps)
The radio module accepts this, as expected.
- After that, downlink sequence counter increased to 65536 (rollover for 32 bits sequence counter)
The radio module rejects this frame

With the change everything works fine.

itziardelatorre committed:
Merge remote-tracking branch 'remotes/origin/master'

itziardelatorre committed:
Applied suggested astyle fix(es)
